src/servingTree/servingDataProvider.ts

```diff
@@ -443,87 +415,57 @@ export class ServingDataProvider implements TreeDataProvider<ServingTreeItem | E
     // *** As a hack, make a file for the yaml, use kubectl apply to create, then delete the file
     // Check the local files for the URL for YAML file
     const serviceName = serveObj.name;
-    let files: [string, FileType][];
+    const tempPath = os.tmpdir();
+    const fsPath = path.join(tempPath, `${serviceName}.yaml`);
+    await fsx.ensureFile(fsPath);
+    const stringContent = yaml.parse(
+      `apiVersion: serving.knative.dev/v1\nkind: Service\nmetadata:\n  name: ${serveObj.name}\nspec:\n  template:\n    spec:\n      containers:\n      - image: ${serveObj.image}`,
+    ) as svc.Items;
+    const yamlContent = yaml.stringify(stringContent);
+
+    await fsx.writeFile(fsPath, yamlContent);
```

Comment:
error handling is different from writeFile.

Reply:
ok but then why you should have a `writeFile` method inside the virtualfs? That's not used anywhere else. 
If the problem is custom error handling you would face it with any other resource. Today service, tomorrow something else. Maybe you should change the `writeFile` method in virtualfs so to be used generally with any other resource in future.  Return the ClIExitData or something else so that the calling method can add a custom error handling, or make it accepts a callback, no?

Reply:
I have change the code :)